controllers/keda/triggerauthentication_controller.go

```diff
@@ -66,7 +66,9 @@ func (r *TriggerAuthenticationReconciler) Reconcile(ctx context.Context, req ctr
 		if errors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		reqLogger.Error(err, "Failed to get TriggerAuthentication")
+		msg := "Failed to get TriggerAuthentication"
+		reqLogger.Error(err, msg)
+		r.EventRecorder.Event(triggerAuthentication, corev1.EventTypeNormal, eventreason.TriggerAuthenticationFailed, msg)
```

Comment:
Will this work? I mean, we have failed getting the resource, will we be able to update it or we'll receive a panic?